vassal-app/src/main/java/VASSAL/script/ExpressionInterpreter.java

```diff
@@ -269,23 +269,18 @@ public String evaluate(PropertySource ps, boolean localized) throws ExpressionEx
         if (value == null) {
           setVar(var, "");
         }
-        else if ("true".equals(value)) {
+        else if (BeanShell.TRUE.equals(value)) {
           setVar(var, true);
         }
-        else if ("false".equals(value)) {
+        else if (BeanShell.FALSE.equals(value)) {
           setVar(var, false);
         }
         else {
           try {
             setVar(var, Integer.parseInt(value));
           }
           catch (NumberFormatException e) {
-            try {
-              setVar(var, Float.parseFloat(value));
-            }
-            catch (NumberFormatException e1) {
-              setVar(var, value);
-            }
```

Comment:
This should be the only actual code change to prevent "1D" and "1F" being converted to "1.0". The Float.parseFloat() check should never have been there.